internal/oidc/auth/auth_handler.go

```diff
@@ -59,7 +59,11 @@ func NewHandler(
 		}
 
 		if oidcUpstream != nil {
-			return handleAuthRequestForOIDCUpstream(r, w,
+			if len(r.Header.Values(CustomUsernameHeaderName)) > 0 {
```

Comment:
They are both checked for non-emptiness by the `requireNonEmptyUsernameAndPasswordHeaders()` helper.

Not sure where to put those constants in the `apis` package. I created `apis/idpdiscovery/v1alpha1` for all the Supervisor discovery endpoint stuff. These would not fit into that package since they are for the authorize endpoint and are not versioned. Maybe a new package? This new package would probably only ever have string constants in it.

Reply:
I put the string constants for the authorize endpoint into `apis/supervisor/oidc` in 04b8f0b45595ccf4e60b6a78625b2125d81d5a0e.